app/controllers/api/v1/organizations_controller.rb

```diff
@@ -2,8 +2,91 @@ module Api
   module V1
     class OrganizationsController < ApiController
       include Api::OrganizationsController
-
       before_action :find_organization, only: %i[users listings articles]
+      before_action :authenticate!, only: %i[create update destroy]
+      before_action :authorize_admin, only: %i[create update]
+      before_action :authorize_super_admin, only: %i[destroy]
+      after_action :verify_authorized, only: %i[create update destroy]
+
+      INDEX_ATTRIBUTES_FOR_SERIALIZATION = %i[
+        id name profile_image slug summary tag_line url
+      ].freeze
+
+      SHOW_ATTRIBUTES_FOR_SERIALIZATION = %i[
+        id username name summary twitter_username github_username url
+        location created_at profile_image slug tech_stack tag_line story
+      ].freeze
+
+      def index
+        per_page = (params[:per_page] || 30).to_i
+        num = [per_page, per_page_max].min
+        page = params[:page] || 1
+
+        organizations = Organization.select(INDEX_ATTRIBUTES_FOR_SERIALIZATION).page(page).per(num)
+
+        render json: organizations
+      end
+
+      def show
+        # This is less RESTful than having a show action that would always assume an id
+        # unless given e.g. a query parameter specifying a different lookup key.
+        # The by-username lookup is the current default behavior, this approach
+        # keeps it intact. Conventionally, the lookup would default to id-based if we keep both.
+        finder = Organization.select(SHOW_ATTRIBUTES_FOR_SERIALIZATION)
+        @organization = finder.find_by(id: params[:id_or_slug]) || finder.find_by(slug: params[:id_or_slug])
+        raise ActiveRecord::RecordNotFound unless @organization
+
+        render :show
+      rescue ArgumentError => e
+        render json: { error: e }, status: :unprocessable_entity
+      end
+
+      def update
+        set_organization
+        @organization.assign_attributes(organization_params)
+        if @organization.save
+          render json: {
+            id: @organization.id,
+            name: @organization.name,
+            profile_image: @organization.profile_image,
```

Comment:
Correct me if I'm not reading this properly, but I don't think `profile_image` is the actual Rails method we want to be calling because this is an object. We'll want `profile_image_url`.

With carrierwave, the getter for the url is `profile_image_url` and to set a profile image from url it would be `remote_profile_image_url = `.

Reply:
:link: https://github.com/carrierwaveuploader/carrierwave/blob/master/README.md#uploading-files-from-a-remote-location

Reply:
an oversight on my part! thanks!